### PR TITLE
The kubernetes test controller will now fetch logs and save them before tearing down a container (#2063)

### PR DESCRIPTION
**Pull request by wcs1only**
# Description

We'll now pull logs for every container and upload them as a workflow artifact.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2063

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/platforms/kubernetes/appmanager.go [hunk 1]
```diff
@@ -428,6 +454,58 @@ func (m *AppManager) GetHostDetails() (string, string, error) {
 	return podList.Items[0].GetName(), podList.Items[0].Status.PodIP, nil
 }
 
+// SaveContainerLogs get container logs for all containers in the pod and saves them to disk
+func (m *AppManager) SaveContainerLogs() error {
```
Comment (mukundansundar) [inline tests/platforms/kubernetes/appmanager.go:458]: if possible please add tests for the same.

Comment (wcs1only) [reply tests/platforms/kubernetes/appmanager.go:458]: Fair, though tests for our tests is kind of meta. :)

Comment (mukundansundar) [reply tests/platforms/kubernetes/appmanager.go:458]: I accept that tests for tests is meta :) But just saw that appmanager_test.go exists. So only asked :)

Comment (wcs1only) [reply tests/platforms/kubernetes/appmanager.go:458]: No you're right, no reason not to test this.

Comment (wcs1only) [reply tests/platforms/kubernetes/appmanager.go:458]: Ok, so I went down this road. To test SaveContainerLogs using the kubernetes client mocks, we need the v0.19.0 golang kubernetes client. Specifically we need https://github.com/kubernetes/client-go/commit/cacbf4f9e0c9bfc9255741e40eafa96c630e26b5 . Upgrading broke a bunch of other things, so I don't think it is worth spending any more time trying to unit test part of our e2e test fixture.

**Review by youngbupark — APPROVED** [review]
👍

File: tests/platforms/kubernetes/appmanager.go (hunk 1 above)
Comment (youngbupark) [reply tests/platforms/kubernetes/appmanager.go:458]: we can revisit later... :) investigating e2e test is more important now...

**Pull request merged**